pkg/reconciler/v1alpha1/build/build.go

```diff
@@ -86,25 +88,40 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 		return nil
 	}
 
-	pod, err := c.reconcileBuildPod(build)
-	if err != nil {
-		return err
-	}
+	if build.Rebasable() {
+		//todo how to handle errors
 
-	if build.MetadataReady(pod) {
-		image, err := c.MetadataRetriever.GetBuiltImage(build)
+		image, err := c.ImageRebaser.Rebase(build)
 		if err != nil {
 			return err
 		}
 
 		build.Status.BuildMetadata = buildMetadataFromBuiltImage(image)
 		build.Status.LatestImage = image.Identifier
-	}
+		build.Status.RunImage = image.RunImage
+	} else {
+
+		pod, err := c.reconcileBuildPod(build)
+		if err != nil {
+			return err
+		}
+
+		if build.MetadataReady(pod) {
+			image, err := c.MetadataRetriever.GetBuiltImage(build)
+			if err != nil {
+				return err
+			}
 
-	build.Status.PodName = pod.Name
-	build.Status.StepStates = stepStates(pod)
-	build.Status.StepsCompleted = stepCompleted(pod)
-	build.Status.Conditions = conditionForPod(pod)
+			build.Status.BuildMetadata = buildMetadataFromBuiltImage(image)
+			build.Status.LatestImage = image.Identifier
+			build.Status.RunImage = image.RunImage
+		}
+
+		build.Status.PodName = pod.Name
+		build.Status.StepStates = stepStates(pod)
+		build.Status.StepsCompleted = stepCompleted(pod)
```

Comment:
Does it make sense to have step states for rebase?

Reply:
Indentation makes it look weird, but the rebase flow skips those lines.

Reply:
Yeah, I was more asking if we _should_ have step states on the rebase build?

It wouldn't need to be in this PR.

Reply:
It makes sense if/when it happens in its own pod. That is something to think about though.